pymc_marketing/mmm/base.py

```diff
@@ -472,19 +481,491 @@ def compute_channel_contribution_original_scale(self) -> DataArray:
+    def _estimate_budget_contribution_fit(
+        self, method: str, channel: str, budget: float
+    ) -> Tuple:
+        """
+        Estimate the lower and upper bounds of the contribution fit for a given channel and budget.
+        This function computes the quantiles (0.05 & 0.95) of the channel contributions, estimates the parameters of the fit function based on the specified method (either 'sigmoid' or 'michaelis-menten'), and calculates the lower and upper bounds of the contribution fit.
+        The function is used in the `plot_budget_scenearios` function to estimate the contribution fit for each channel and budget scenario.
+        The estimated fit is then used to plot the contribution optimization bounds for each scenario.
+
+        Parameters
+        ----------
+        method : str
+            The method used to fit the contribution & spent non-linear relationship. It can be either 'sigmoid' or 'michaelis-menten'.
+        channel : str
+            The name of the channel for which the contribution fit is being estimated.
+        budget : float
+            The budget for the channel.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the lower and upper bounds of the contribution fit.
+
+        Raises
+        ------
+        ValueError
+            If the method is not 'sigmoid' or 'michaelis-menten'.
+        """
+        channel_contributions_quantiles = (
+            self.compute_channel_contribution_original_scale().quantile(
+                q=[0.05, 0.95], dim=["chain", "draw"]
+            )
+        )
+
+        # Estimate parameters based on the method
+        if method == "sigmoid":
+
+            alpha_limit_upper, lam_constant_upper = estimate_sigmoid_parameters(
+                channel, self.X, channel_contributions_quantiles.sel(quantile=0.95)
+            )
+            alpha_limit_lower, lam_constant_lower = estimate_sigmoid_parameters(
+                channel, self.X, channel_contributions_quantiles.sel(quantile=0.05)
+            )
+
+            fit_function = extense_sigmoid
+        elif method == "michaelis-menten":
+
+            alpha_limit_upper, lam_constant_upper = estimate_menten_parameters(
+                channel, self.X, channel_contributions_quantiles.sel(quantile=0.95)
+            )
+            alpha_limit_lower, lam_constant_lower = estimate_menten_parameters(
+                channel, self.X, channel_contributions_quantiles.sel(quantile=0.05)
+            )
+
+            fit_function = michaelis_menten
+        else:
+            raise ValueError("`method` must be either 'michaelis-menten' or 'sigmoid'.")
+
+        y_fit_lower = fit_function(budget, alpha_limit_lower, lam_constant_lower)
+        y_fit_upper = fit_function(budget, alpha_limit_upper, lam_constant_upper)
+
+        return y_fit_lower, y_fit_upper
+
+    def _standardize_scenarios_dict_keys(self, d: Dict, keywords: list):
```

Comment:
IMO, this should be in a utils module as it does not depend on the MMM class.

- The type hint should be `keywords: List[str]`